# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/IonBuilder.h
# Commit: 3d8cfd4a935d
# Full Hash: 3d8cfd4a935dad17ab09a64325de2101062eaf1d
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-26 16:16:58
# Regressor Bug: 1599117
# File Overlap Count: 1
# Description:
#   Bug 1599117 part 3 - Remove IonBuilder::detectAndOrStructure. r=tcampbell
#   
#   I think this has been effectively dead code for a few years because we no longer
#   create the triangle structure for JSOP_AND and JSOP_OR.
#   
# ==============================================================================

diff -r 2c25a18171ba -r 3d8cfd4a935d js/src/jit/IonBuilder.h
--- a/js/src/jit/IonBuilder.h	Tue Nov 26 11:27:59 2019 +0000
+++ b/js/src/jit/IonBuilder.h	Mon Nov 25 17:59:22 2019 +0000
@@ -364,8 +364,6 @@
   AbortReasonOr<Ok> improveTypesAtTypeOfCompare(MCompare* ins, bool trueBranch,
                                                 MTest* test);
 
-  // Used to detect triangular structure at test.
-  bool detectAndOrStructure(MPhi* ins, bool* branchIsTrue);
   AbortReasonOr<Ok> replaceTypeSet(MDefinition* subject, TemporaryTypeSet* type,
                                    MTest* test);
 
